x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_upgrade_modal/index.tsx

```diff
@@ -248,6 +268,31 @@ export const AgentUpgradeAgentModal: React.FunctionComponent<Props> = ({
           }}
         />
       </EuiFormRow>
+      {isScheduled && (
+        <>
+          <EuiSpacer size="m" />
+          <EuiFormRow
+            label={i18n.translate('xpack.fleet.upgradeAgents.startTimeLabel', {
+              defaultMessage: 'Scheduled date and time',
+            })}
+            fullWidth
+          >
+            <EuiDatePicker
```

Comment:
could the displayed datetime be formatted more user friendly?

Reply:
> * the schedule action is only visible as bulk action, is it intentional?
Yes, this is per requirements. The schedule upgrade is intended to be a separate action from the bulk upgrade, and it's only for the bulk upgrade because it's mostly needed when the user needs to upgrade many agents.

> * nit: do we need to show seconds as well or only minutes?

I checked the Figma designs for this feature and it seems ok. No seconds:
![Screenshot 2022-05-23 at 15 37 49](https://user-images.githubusercontent.com/16084106/169832056-07f1e2af-3e7e-4229-aa07-7c53705615db.png)

> * the default start time appears to be quite random, I suppose it is the time when the server was started?
Yes I think so, it's set to `moment()`. I'll see if I can change it to something more meaningful.

> * the format is still a bit weird, it says 15:05PM - it should be either 15 or 3PM
> * the values in the datetime picker appear to be duplicated

I'll check why, there's something misconfigured in the Datepicker probably